Source/Core/Core/HW/DVD/AMMediaboard.cpp

```diff
@@ -0,0 +1,1844 @@
+  if (!s_netctrl.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "trinetctrl.bin");
+  if (!s_extra.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "triextra.bin");
+  if (!s_dimm.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "tridimm.bin");
+  if (!s_backup.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "backup.bin");
+
+  // This is the firmware for the Triforce
+  const std::string sega_boot_filename = base_path + "segaboot.gcm";
+
+  if (!File::Exists(sega_boot_filename))
+  {
+    PanicAlertFmt("Failed to open segaboot.gcm({}), which is required for test menus.",
+                  sega_boot_filename.c_str());
+    return;
+  }
+
+  File::IOFile sega_boot(sega_boot_filename, "rb+");
+  if (!sega_boot.IsOpen())
+  {
+    PanicAlertFmt("Failed to read: {}", sega_boot_filename);
+    return;
+  }
+
+  const u64 length = std::min<u64>(sega_boot.GetSize(), sizeof(s_firmware));
+  sega_boot.ReadBytes(s_firmware, length);
+
+  s_test_menu = true;
+}
+
+u8* InitDIMM(u32 size)
+{
+  if (size == 0)
+    return nullptr;
+
+  if (s_dimm_disc.empty())
+  {
+    s_dimm_disc.reset(size);
+    if (s_dimm_disc.empty())
+    {
+      PanicAlertFmt("Failed to allocate DIMM memory.");
+      return nullptr;
+    }
+  }
+
+  s_firmware_map = false;
+  return s_dimm_disc.data();
+}
+
+static s32 NetDIMMAccept(int fd, sockaddr* addr, socklen_t* len)
+{
+  SOCKET client_sock = INVALID_SOCKET;
+  fd_set readfds;
+
+  FD_ZERO(&readfds);
+  FD_SET(fd, &readfds);
+
+  timeval timeout{
+      .tv_sec = 0,
+      .tv_usec = 10000,  // 10 milliseconds
+  };
+
+  const int result = select(fd + 1, &readfds, nullptr, nullptr, &timeout);
+  if (result > 0 && FD_ISSET(fd, &readfds))
+  {
+    client_sock = accept_(fd, addr, len);
+    if (client_sock != INVALID_SOCKET)
+    {
+      s_last_error = SSC_SUCCESS;
+      return client_sock;
+    }
+
+    s_last_error = SOCKET_ERROR;
+    return SOCKET_ERROR;
+  }
+
+  if (result == 0)
+  {
+    // Timeout
+    s_last_error = SSC_EWOULDBLOCK;
+  }
+  else
+  {
+    // select() failed
+    s_last_error = SOCKET_ERROR;
+  }
+
+  return SOCKET_ERROR;
+}
+
+static s32 NetDIMMConnect(int fd, sockaddr_in* addr, int len)
+{
+  // All.Net Connect IP
+  if (addr->sin_addr.s_addr == inet_addr("192.168.150.16"))
+  {
+    addr->sin_addr.s_addr = inet_addr("127.0.0.1");
+  }
+
+  // CyCraft Connect IP
+  if (addr->sin_addr.s_addr == inet_addr("192.168.11.111"))
+  {
+    addr->sin_addr.s_addr = inet_addr("127.0.0.1");
+  }
+
+  // NAMCO Camera ( IPs are: 192.168.29.104-108 )
+  if ((addr->sin_addr.s_addr & 0xFFFFFF00) == 0xC0A81D00)
+  {
+    addr->sin_addr.s_addr = inet_addr("127.0.0.1");
+
+    // BUG: An invalid family value is being used
+    addr->sin_family = htons(AF_INET);
+    s_namco_cam = fd;
+  }
+
+  // Key of Avalon Client
+  if (addr->sin_addr.s_addr == inet_addr("192.168.13.1"))
+  {
+    addr->sin_addr.s_addr = inet_addr("10.0.0.45");
```

Comment:
The other addresses get converted to loopback, is there a reason this one does not?

Reply:
I believe on some platforms, you can use 127.0.0.1 and 127.0.0.2 as separate loopback addresses (anything starting with 127, even), although user-selectable addresses would probably be better.